src/ess/reduce/live/raw.py

```diff
@@ -317,19 +289,64 @@ def make_roi_filter(self) -> roi.ROIFilter:
                 indices = self._projection(indices)
         return roi.ROIFilter(indices=indices, norm=norm)
 
+    def transform_weights(
+        self,
+        weights: sc.Variable | sc.DataArray | None = None,
+        *,
+        threshold: float = 0.1,
+    ) -> sc.DataArray:
+        """
+        Transform raw pixel weights to the projection plane.
+
+        Parameters
+        ----------
+        weights:
+            Raw pixel weights to transform. If None, default weights of 1 are used.
+        threshold:
+            Threshold for identifying bins with a low weight. If the weight is below the
+            threshold times the median weight, the bin is marked as invalid. This is
+            relevant to avoid issues with color scales in plots, where noise in bins
+            with low weight may dominate the color scale if auto-scaling is used.
+        """
+        if weights is None:
+            weights = sc.ones(
+                sizes=self.detector_number.sizes, dtype='float32', unit=''
+            )
+        else:
+            if weights.sizes != self.detector_number.sizes:
+                raise sc.DimensionError(
+                    f'Invalid {weights.sizes=} for {self.detector_number.sizes=}.'
+                )
+            if (
+                isinstance(weights, sc.DataArray)
+                and (detector_number := weights.coords.get('detector_number'))
+                is not None
+            ):
+                if not sc.identical(detector_number, self.detector_number):
+                    raise sc.CoordError("Mismatching detector numbers in weights.")
+                weights = weights.data
+        if isinstance(self._projection, Histogrammer):
+            xs = self._projection.apply_full(weights)  # Use all replicas
+        elif self._projection is not None:
+            xs = self._projection(weights)
+        else:
+            xs = weights.copy()
```

Comment:
This now tightly couples `Logicalview` to `Histogrammer`.

I am unclear about what a projection is here. Does it know about replicas in general? Or does only `Histogrammer` know about them? This looks to me like it would lead to inconsistent results between different projections. Can `apply_full` be part of the projection protocol?

Reply:
> This now tightly couples `Logicalview` to `Histogrammer`.

How so?

> I am unclear about what a projection is here. Does it know about replicas in general? Or does only `Histogrammer` know about them? This looks to me like it would lead to inconsistent results between different projections. Can `apply_full` be part of the projection protocol?

Only the histogrammer. Replicas are not needed in other cases. In what sense inconsistent?

> Can `apply_full` be part of the projection protocol?

It could, but testing is a bit more cumbersome since I cannot pass a simple function as projection (in existing tests unrelated to this change).

Reply:
Sorry, not `LogicalView` but `RollingDetectorView`. I unfolded the wrong lines in the diff.

Ok. I am worried that this becomes confusing and complicated as the code grows. But if you can keep it contained and an implementation detail, it should be fine.